website/blog/2019-02-01-tin.md

```diff
@@ -0,0 +1,96 @@
+---
+author: Ólafur Páll Geirsson
+title: Metals v0.4.1
+authorURL: https://twitter.com/olafurpg
+authorImageURL: https://avatars2.githubusercontent.com/u/1408093?s=460&v=4
+---
+
+Metals is a language server for Scala that works with VS Code, Atom, Vim,
+Sublime Text and Emacs. Metals is developed at the
+[Scala Center](https://scala.epfl.ch/) along with contributors from the
+community.
+
+<!-- truncate -->
+
+## Metals server
+
+In this milestone we merged 5 PRs, full details:
+https://github.com/scalameta/metals/milestone/5?closed=1.
+
+### Improved code navigation
+
+Several "goto definition" and "find references" bugs have been fixed in this
+release. In particular, code navigation should work more reliably now for the
+following language features
+
+- for comprehensions with multiple assignments and guards
+- eta-expanded methods and functions passed as values
+- val patterns like `val (a, b) = ...`
+- named arguments in `apply` methods
+- repeated parameter types (varargs)
+
+### Empty source directories are no longer created
+
+Previously, Metals created all source directories like `src/main/java` and
+`src/main/scala-2.12` even if they were unused. Now, Metals still creates these
+directories but removes them again after the file watcher has started. Big
+thanks to first-time contributor [@mudsam](https://github.com/mudsam) for
+implementing this fix!
```

Comment:
Maybe it's worth to mention the potential drawbacks of this approach? Like the chance of missing file events if those directories are created after?

Reply:
I believe the cases the current approach will fail are rare enough that it won't make a difference to mention it here. The general troubleshooting advice "Run 'connect to build server'" fixes this issue along with other similar issue (that we're not aware of).

Reply:
Fair enough 👍